test/unit/math/prim/mat/functor/map_rect_mpi_test.cpp

```diff
@@ -3,6 +3,7 @@
 #ifdef STAN_MPI
 
 #include <stan/math/prim/mat.hpp>
+#include <stan/math/rev/mat.hpp>
```

Comment:
I think we should move these files out from `prim` so they are allowed to include `rev`, right?

Reply:
In mpi map rect tests only need the rev part as they callout to map_rect_concurrent whenever the MPI cluster is busy. However, since we will move the `prim` version of map_rect_concurrent back to `prim` once we have a threadpool, I would prefer to leave the files in the place where they are.

Maybe we add a TODO comment there? If you feel strong about this, we can (temporarily) move it.

Reply:
@wds15, this will instantiate the template metaprograms in the incorrect order, so please fix this.

Either move this test to `rev` and just include `rev/mat.hpp` or don't include it at all.

What this will do is instantiate things for `double` and not for `stan::math::var` and this might lead to undefined behavior if you're actually expecting things to work with `stan::math::var`.